src/hooks/useActiveOnboard.ts

```diff
@@ -14,125 +9,117 @@ import web3Utils from 'web3-utils'
+    const error = !isSupported ? new UnsupportedChainId(chainIdHex) : undefined
+
+    const newContext = {
+        active: true,
+        accounts: accounts,
+        chainId: parseInt(chainId),
+        account: web3Utils.toChecksumAddress(accounts[0].address),
+        label: label,
+        eipProvider: provider as EIP1193ProviderExtended,
+        library: web3provider,
+        error: error,
+    }
+    return newContext
 }
 
-export function useActiveOnboard<T = any>():ActiveOnboardInterface<T> {
-  const connectedWallets = useWallets()
-  const context = useMemo<ActiveOnboardInterface<Web3Provider>>(() => {
-    if (connectedWallets.length > 0) {
-      // console.log('update ActiveOnboard context')
-      const newContext = onboardContext(connectedWallets)
-      return newContext
-    } else {
-      return ({active: false, chainId: 122})
-    }
-  }, [connectedWallets])
+export function useActiveOnboard<T = any>(): ActiveOnboardInterface<T> {
+    const connectedWallets = useWallets()
+    const context = useMemo<ActiveOnboardInterface<Web3Provider>>(() => {
+        if (connectedWallets.length > 0) {
+            const newContext = onboardContext(connectedWallets)
+            return newContext
+        } else {
+            return { active: false, chainId: 42220 }
+        }
+    }, [connectedWallets])
 
-  return context
+    return context
 }
 
-
 /** Store connected wallet data in localStorage to be used for eagerly connecting on page-load
  * @param wallets all currently connected wallets
  * @param activeChainId active chain id is the currently connected chain if supported, else it defaults to mainnet
  * @returns void
  */
- export function StoreOnboardState(wallets:WalletState[], activeChainId:string | undefined):void {
-  if (wallets.length === 0){
-    localStorage.removeItem('currentConnectWallet') 
-  } else {
-    const walletLabel = wallets.map(({ label }) => label)
-    const connectedAccount = wallets.map(({accounts}) => accounts[0])
-    const connectedChain = activeChainId     
-    const connected = [{
-      accounts: connectedAccount,
-      chains: connectedChain,
-      label: walletLabel}
-    ]
-    localStorage.setItem(
-      'currentConnectWallet',
-      JSON.stringify(connected)
-    )
-  }
+export function StoreOnboardState(wallets: WalletState[], activeChainId: string | undefined): void {
+    if (wallets.length === 0) {
+        localStorage.removeItem('currentConnectWallet')
+    } else {
+        const walletLabel = wallets.map(({ label }) => label)
+        const connectedAccount = wallets.map(({ accounts }) => accounts[0])
+        const connectedChain = activeChainId
+        const connected = [
+            {
+                accounts: connectedAccount,
+                chains: connectedChain,
+                label: walletLabel,
+            },
+        ]
+        localStorage.setItem('currentConnectWallet', JSON.stringify(connected))
```

Comment:
add some library providing a) some level of abstraction b) json serialization under cover

in wallet app we used RN community's AsyncStorage with own-made serialization
https://github.com/GoodDollar/GoodDAPP/blob/master/src/lib/utils/asyncStorage.js
but it doesn't needed here. Pls find some web-only solution or just adjust the class i've linked to call localStorage instead of AsyncStorage

Reply:
@L03TJ3 done here: https://github.com/GoodDollar/GoodProtocolUI/pull/246/commits/79e1a7b0bab00d9e5c17cbdd95c66a229905d86c

this requires the following PR to be merged onto SDK: https://github.com/GoodDollar/GoodWeb3-Mono/pull/24